src/main/java/SpringWeatherForecast/WeatherForecaster.java

```diff
@@ -0,0 +1,46 @@
+package SpringWeatherForecast;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+
+
+import java.util.Optional;
+
+public class WeatherForecaster {
+	
+	DataSource source = new DataSource();
```

Comment:
This is not good that you have 2 classes, which depends on DataSource. There should be only one. This means you have a wrong design and wrong coupling.